code/modules/mob/transform_procs.dm

```diff
@@ -37,7 +37,9 @@
 	O.add_ai_verbs()
 
 	O.rename_self("AI",1)
-	qdel(src)
+

```

Comment:
I'd hedge bets and add `. = O` in case `qdel` does manage to get run before the return does